File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    if (arr.length === 1) {
        return [-1];
    }

    let minIndex = 0;
    let minValue = arr[0];

    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < minValue) {
            minIndex = i;
            minValue = arr[i];
        }
    }

    arr.splice(minIndex, 1);

    return arr;
}

File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        if (arr.length <= 1) {
            return new int[]{-1};
        }

        int min = Arrays.stream(arr).min().getAsInt(); // 배열에서 최소값 찾기

        ArrayList<Integer> result = new ArrayList<>();

        for (int num : arr) {
            if (num != min) {
                result.add(num);
            }
        }

        int[] answer = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(i);
        }

        return answer;
    }
}
